src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,234 @@
+    """Tests :func:`orthology_benchmark.dump_genomes()` function.
+
+    Attributes:
+        core_dbs: A set of test core databases.
+
+    """
+
+    core_dbs = {}  # type: Dict
+    host = None  # type: str
+    port = None  # type: int
+    username = None  # type: str
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, request: FixtureRequest, multi_dbs: Dict) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            request: Access to the requesting test context.
+            multi_dbs: Dictionary of unit test databases (fixture).
+
+        """
+        type(self).core_dbs = multi_dbs
+        server_url = sqlalchemy.engine.url.make_url(request.config.getoption('server'))
+        type(self).host = server_url.host
+        type(self).port = server_url.port
+        type(self).username = "ensro" if server_url.username == "ensadmin" else server_url.username
+
+    @pytest.mark.skipif(os.environ['USER'] == 'travis',
+                        reason="The test requires both Perl and Python which is not supported by Travis.")
+    @pytest.mark.parametrize(
+        "species_list, species_set_name, expectation",
+        [
+            (["gallus_gallus", "homo_sapiens"], "default", does_not_raise()),
+            (["homo_sapiens", "zea_mays"], "default", raises(FileExistsError)),
+            (["felis_catus", "zea_mays"], "test", raises(RuntimeError,
+                    match=r"No cores found for the species set 'test' on the specified host."))
+        ]
+    )
+    def test_dump_genomes(self, species_list: List[str], species_set_name: str,
+                          tmp_dir: Path, expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` when server connection can be established.
+
+        Args:
+            species_list: A list of species (genome names).
+            species_set_name: Species set (collection) name.
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            orthology_benchmark.dump_genomes(species_list, species_set_name, self.host, self.port,
+                                             self.username, tmp_dir)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "orth_benchmark"  # type: ignore[attr-defined]
+            for db_name, unittest_db in self.core_dbs.items():
+                assert file_cmp(out_files / f"{unittest_db.dbc.db_name}.fasta", exp_out / f"{db_name}.fasta")
```

Comment:
Forgive me if I'm asking this a little late in the review process, but have you considered using the shorter extension `.fa` for the FASTA files? With the shorter extension, this line would be a little less of a squeeze.

Reply:
I came across `.fasta` files on Noah/Codon so decided to keep it long. The extension actually comes form `orthology_benchmark.py`, line 72.

Reply:
There are also examples of files using `.fa` to be found (e.g. [here](https://ftp.ensembl.org/pub/release-104/fasta/), but both extensions are perfectly fine, so I'll leave it up to you.

Reply:
Moved to `.fa`. `pytest` passes locally.